src/frameworks/next/utils.ts

```diff
@@ -209,25 +212,80 @@ export async function isUsingMiddleware(dir: string, isDevMode: boolean): Promis
  */
-export async function isUsingImageOptimization(dir: string): Promise<boolean> {
-  let { isNextImageImported } = await readJSON<ExportMarker>(join(dir, EXPORT_MARKER));
+export async function isUsingImageOptimization(
+  projectDir: string,
+  distDir: string
+): Promise<boolean> {
+  let { isNextImageImported } = await readJSON<ExportMarker>(join(distDir, EXPORT_MARKER));
   // App directory doesn't use the export marker, look it up manually
-  if (!isNextImageImported && isUsingAppDirectory(dir)) {
-    isNextImageImported = (await readFile(join(dir, "server", "client-reference-manifest.js")))
-      .toString()
-      .includes("node_modules/next/dist/client/image.js");
+  if (!isNextImageImported && isUsingAppDirectory(distDir)) {
+    if (await isUsingNextImageInAppDirectory(projectDir, distDir)) {
+      isNextImageImported = true;
+    }
   }
 
   if (isNextImageImported) {
-    const imagesManifest = await readJSON<ImagesManifest>(join(dir, IMAGES_MANIFEST));
+    const imagesManifest = await readJSON<ImagesManifest>(join(distDir, IMAGES_MANIFEST));
     return !imagesManifest.images.unoptimized;
   }
 
   return false;
 }
 
+/**
+ * Whether next/image is being used in the app directory
+ */
+async function isUsingNextImageInAppDirectory(
+  projectDir: string,
+  nextDir: string
+): Promise<boolean> {
+  function fileIncludesNextImageComponent(fileContents: string): boolean {
+    return fileContents.includes("node_modules/next/dist/client/image");
+  }
+
+  // Note: casting the types here as this function is called after the Next.js build is done,
+  // so Next.js was already identified as a dependency. In this case getNextVersion
+  // will always return a valid string
+  const nextVersion = getNextVersion(projectDir) as string;
+  const nextVersionSemver = minVersion(nextVersion) as SemVer;
+
+  // Note: canary Next.js versions e.g. 13.4.10-canary.0 are identified as lower than 13.4.10 by SemVer.
+  // Here we use only the major, minor and patch versions for comparison.
+  const nextVersionString = `${nextVersionSemver.major}.${nextVersionSemver.minor}.${nextVersionSemver.patch}`;
+
+  if (gte(nextVersionString, "13.4.10")) {
+    // Next.js >= 13.4.10 has one client-reference-manifest file per entry
+    const files = await new Promise<string[]>((resolve) => {
+      glob(
+        join(nextDir, "server", "app", "**", "*client-reference-manifest.js"),
+        (err, matches) => {
+          resolve(matches);
+        }
+      );
+    });
+
+    // Return true when the first file containing the next/image component is found
+    for (const filepath of files) {
+      const fileContents = readFileSync(filepath);
+      if (fileIncludesNextImageComponent(fileContents.toString())) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  // Next.js < 13.4.10 has a single client-reference-manifest.js file
+  const clientReferenceManifest = await readFile(
```

Comment:
Move this logic (`Next.js < 13.4.10`) above the more complex `Next.js >= 13.4.10` logic.
So it's in chronological order and from simple to complex.

Reply:
good call, thanks!